libraries/wavemap/include/wavemap/integrator/measurement_model/impl/continuous_ray_inl.h

```diff
@@ -48,21 +48,22 @@ inline FloatingPoint ContinuousRay::computeUpdate(
 }
 
 inline FloatingPoint ContinuousRay::computeBeamUpdate(
-    const Vector3D& sensor_coordinates, const Index2D& image_index) const {
+    const SensorCoordinates& sensor_coordinates,
+    const Index2D& image_index) const {
   if (!range_image_->isIndexWithinBounds(image_index)) {
     return 0.f;
   }
 
   const FloatingPoint measured_distance = range_image_->at(image_index);
 
-  if (range_threshold_back_ < sensor_coordinates.z() - measured_distance) {
+  if (range_threshold_back_ < sensor_coordinates.normal - measured_distance) {
```

Comment:
Normal might be an odd name given that one would assume this to be a 2D or 3D vector depending on the space.

Reply:
This coordinate represents `depth` for cameras and `range` for LiDARs. The most intuitive and correct name I could think of was `normal`, as it's perpendicular to the image surface for both sensors (plane for cams, dome for LiDARs). Do you perhaps have a better idea? :) Otherwise we could also just call it `depth` (or `range`) and add an explanation to the type declaration.

Reply:
The full type is currently
```
using ImageCoordinates = Vector2D;
struct SensorCoordinates {
  ImageCoordinates image;
  FloatingPoint normal;
};
```
which leads to this access syntax `sensor_coordinates.image.x()`, `sensor_coordinates.image.y()` and `sensor_coordinates.normal`; and also automatically provides methods like `sensor_coordinates.image.norm()`.

Reply:
Yep I remembered that which is why the normal bit was somewhat odd to me. As mentioned the first thing that comes to my mind when I read "normal" is a normal vector as opposed to a scalar. So I guess the depth name you suggested makes the most sense to me.

Reply:
Switched it to `depth` as discussed :)